src/components/OptionsSelector.js

```diff
@@ -123,92 +221,94 @@ class OptionsSelector extends Component {
+        if (this.unsubscribeEnter) {
+            this.unsubscribeEnter();
+        }
 
-                    // Wrap around to the top of the list
-                    if (newFocusedIndex > allOptions.length - 1) {
-                        newFocusedIndex = 0;
-                    }
+        if (this.unsubscribeCTRLEnter) {
+            this.unsubscribeCTRLEnter();
+        }
+    }
 
-                    const {index, sectionIndex} = allOptions[newFocusedIndex];
-                    this.scrollToFocusedIndex(sectionIndex, index);
-                    return {focusedIndex: newFocusedIndex};
+    /**
+     * Flattens the sections into a single array of options.
+     * Each object in this array is enhanced to have:
+     *
+     *   1. A `sectionIndex`, which represents the index of the section it came from
+     *   2. An `index`, which represents the index of the option within the section it came from.
+     *
+     * @returns {Array<Object>}
+     */
+    flattenSections() {
+        const allOptions = [];
+        _.each(this.props.sections, (section, sectionIndex) => {
+            _.each(section.data, (option, optionIndex) => {
+                allOptions.push({
+                    ...option,
+                    sectionIndex,
+                    index: optionIndex,
                 });
+            });
+        });
+        return allOptions;
+    }
 
-                e.preventDefault();
-                break;
-            }
+    /**
+     * @param {Number} index
+     */
+    updateFocusedIndex(index) {
+        this.setState({focusedIndex: index}, () => this.scrollToIndex(index));
+    }
 
-            case 'ArrowUp': {
-                this.setState((prevState) => {
-                    let newFocusedIndex = prevState.focusedIndex - 1;
+    /**
+     * Scrolls to the focused index within the SectionList
+     *
+     * @param {Number} index
+     */
+    scrollToIndex(index) {
+        const option = this.state.allOptions[index];
+        if (!this.list || !option) {
+            return;
+        }
 
-                    // Wrap around to the bottom of the list
-                    if (newFocusedIndex < 0) {
-                        newFocusedIndex = allOptions.length - 1;
-                    }
+        const itemIndex = option.index;
+        const sectionIndex = option.sectionIndex;
 
-                    const {index, sectionIndex} = allOptions[newFocusedIndex];
-                    this.scrollToFocusedIndex(sectionIndex, index);
-                    return {focusedIndex: newFocusedIndex};
-                });
-                e.preventDefault();
-                break;
+        // Note: react-native's SectionList automatically strips out any empty sections.
+        // So we need to reduce the sectionIndex to remove any empty sections in front of the one we're trying to scroll to.
+        // Otherwise, it will cause an index-out-of-bounds error and crash the app.
+        let adjustedSectionIndex = sectionIndex;
```

Comment:
SectionList does not appear to be removing the empty sections so this caused this issue #34112. This has been fixed with removing this adjustment of index.